src/Compilers/CSharp/Portable/Symbols/Tuples/TupleTypeSymbol.cs

```diff
@@ -393,7 +393,7 @@ internal static void ReportNamesMismatchesIfAny(ImmutableArray<string> destinati
                 var sourceName = sourceNames[i];
                 if (sourceName != null && (allMissing || string.CompareOrdinal(destinationNames[i], sourceName) != 0))
                 {
-                    diagnostics.Add(ErrorCode.WRN_TupleLiteralNameMismatch, literal.Arguments[i].Syntax.Location, sourceName);
+                    diagnostics.Add(ErrorCode.WRN_TupleLiteralNameMismatch, literal.Arguments[i].Syntax.Parent?.Location, sourceName);
```

Comment:
`Parent` should never be null. And it looks like `diagnostics.Add()` requires a non-null `location` anyway.